ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/data/realtime/RealtimeConversationHelperTest.java

```diff
@@ -0,0 +1,360 @@
+@PrepareForTest({
+        KreConnectionFactory.class,
+        TextUtils.class,
+        KayakoLogHelper.class,
+        KreStarterFactory.class,
+        MessengerOpenTracker.class,
+        RealtimeConversationHelper.class,
+        LoadResourceHelper.class,
+        Handler.class
+})
+public class RealtimeConversationHelperTest {
+
+    private static final long USER_ID = 1L;
+    private static final long CONVERSATION_ID = 1L;
+    private static final int ENTRY_TIME = 0;
+    private static final long MESSAGE_ID = 1L;
+    private static final String USER_AVATAR = "userAvatar";
+    private static final String USER_NAME = "userName";
+    private static final String CHANNEL_NAME = "channelName";
+    private static final boolean IS_TYPING = true;
+    private static final String ERROR_MESSAGE = "exceptionMessage";
+    private static final KayakoException kayakoException = new KayakoException(ErrorCode.OTHER, ERROR_MESSAGE,
+            new Throwable());
+    private static final boolean AUTO_DISABLE_TYPING = true;
+    @Mock
+    private KreStarter kreStarter;
+    @Mock
+    private Handler handler;
+    @Mock
+    private KreCaseSubscription kreCaseSubscription;
+    @Mock
+    private KreConnection kreConnection;
+    @Mock
+    private OnConversationUserOnlineListener onConversationUserOnlineListener;
+    @Mock
+    private OnConversationChangeListener onConversationChangeListener;
+    @Mock
+    private Message message;
+    @Mock
+    private Change change;
+    @Mock
+    private Conversation conversation;
+    @Captor
+    private ArgumentCaptor<OnSubscriptionListener> onSubscriptionListenerCaptor;
+    @Captor
+    private ArgumentCaptor<RawCaseChangeListener> rawCaseChangeListenerCaptor;
+    @Captor
+    private ArgumentCaptor<MinimalClientTypingListener> minimalClientTypingListenerArgumentCaptor;
+    @Captor
+    private ArgumentCaptor<RawCasePostChangeListener> rawCasePostChangeListenerCaptor;
+    @Captor
+    private ArgumentCaptor<RawUserOnCasePresenceListener> rawUserOnCasePresenceListenerCaptor;
+    @Captor
+    private ArgumentCaptor<Runnable> runnableCaptor;
+    @Captor
+    private ArgumentCaptor<ItemCallback<Message>> itemCallbackCaptor;
+    @Captor
+    private ArgumentCaptor<ItemCallback<Conversation>> conversationItemCallbackCaptor;
+    @Captor
+    private ArgumentCaptor<Conversation> conversationCaptor;
+    @Mock
+    private OnConversationClientActivityListener onConversationClientActivityListener;
+    @Captor
+    private ArgumentCaptor<UserViewModel> userViewModelCaptor;
+    @Captor
+    private ArgumentCaptor<Message> messageCaptor;
+    @Captor
+    private ArgumentCaptor<Long> userIdCaptor;
+    @Captor
+    private ArgumentCaptor<Long> conversationIdCaptor;
+    @Captor
+    private ArgumentCaptor<Boolean> isTypingCaptor;
+    @Mock
+    private OnConversationMessagesChangeListener onConversationMessagesChangeListener;
+    private final ErrorCollector errorCollector = new ErrorCollector();
```

Comment:
🚫  This test doesn't contain a mechanism to assert the expected results or exceptional cases. This kind of tests creates artificial code coverage. To consider a test as valid, it must be prone to fail and it must have either assertions or expected exceptions. Consider adding assertions or expect specific exceptions in order to follow testing best practices.  

**[More details](https://confluence.devfactory.com/x/AxqkEw)**
          
## The solution in this case
Add the missing `@Rule` annotation
```JAVA
@Rule
private final ErrorCollector errorCollector = new ErrorCollector();

```
 

<a id="193e3800-4ee7-11e8-b617-bd6867d59d97" href="none" codereview-uuid-reserved-tag />